controllers/configuration/backend/backend.go

```diff
@@ -54,49 +55,20 @@ type Backend interface {
 }
 
 type backendInitFunc struct {
-	initFuncFromHCL  func(*ParsedBackendConfig, client.Client) (Backend, error)
-	initFuncFromConf func(interface{}, client.Client) (Backend, error)
-}
-
-// ParsedBackendConfig is a struct parsed from the backend hcl block
-type ParsedBackendConfig struct {
-	// Name is the label of the backend hcl block
-	// It means which backend type the configuration will use
-	Name string `hcl:"name,label"`
-	// Attrs are the key-value pairs in the backend hcl block
-	Attrs hcl.Body `hcl:",remain"`
-}
-
-func (conf ParsedBackendConfig) getAttrValue(key string) (*cty.Value, error) {
-	attrs, diags := conf.Attrs.JustAttributes()
-	if diags.HasErrors() {
-		return nil, diags
-	}
-	attr := attrs[key]
-	if attr == nil {
-		return nil, fmt.Errorf("cannot find attr %s", key)
-	}
-	v, diags := attr.Expr.Value(nil)
-	if diags.HasErrors() {
-		return nil, diags
-	}
-	return &v, nil
-}
-
-func (conf ParsedBackendConfig) getAttrString(key string) (string, error) {
-	v, err := conf.getAttrValue(key)
-	if err != nil {
-		return "", err
-	}
-	result := ""
-	err = gocty.FromCtyValue(*v, &result)
-	return result, err
+	initFuncFromHCL  func(ctx k8sContext, backendConfig *ParsedBackendConfig, optionSource *OptionSource) (Backend, error)
```

Comment:
For Terraform, HCL and Conf are same to referring to Terraform modules/resource.